fusion-engine-driver/utils/conversion_utils.hpp

```diff
@@ -118,6 +119,21 @@ class ConversionUtils {
     return checksum;
   }
 
+  static int getLastLeapSecondDate() {
+      std::string lastDate;
+      int lastLeapSecond = 0;
+      
+      for (const auto& entry : getLeapSeconds()) {
```

Comment:
This is wrong.
So essentially, lets say we want to plan a few years ahead we don't want to use a leap second in the future if this buffer contains data for 2024 etc...
I'd recommend changing the defition:
getLastLeapSecondDate(size_t year);

Also,
instead of iterating of each, you can probably do a:
`leap_second = leap_seconds_.lower_bounds(current_year);`
Which will return the lower bound of the closest results.